doc/features/pre-existing-tenant.md

```diff
@@ -0,0 +1,59 @@
+# Pre-existing Tenant Feature
+
+# Table of Contents
+
+- [1. Overview](#1-overview)
+- [2. Motivation](#2-motivation)
+- [3. Mechanism](#3-mechanism)
+- [4. Lifecycle](#4-lifecycle)
+- [5. Example](#5-example)
+
+## 1. Overview
+
+This feature is introduced to enable adding a Kubernetes cluster to a pre-existing tenant like the common tenant on APIC. The APIC objects provisioned for different clusters should be uniquely named and distinguishable. When deleting a cluster, the objects created by other clusters shouldn’t be affected.
+
+## 2. Motivation
+
+The pre-existing tenant feature be able to process the tenant field in ACI config of the input yaml file to provision the Kubernetes cluster under the given tenant. This includes creating create application profile, EPGs, BDs and required contracts under the existing tenant. If this tenant isn’t already created, our tool will create it, but not delete it in the un-provision step. All other APIC objects created for this cluster being un-provisioned will be deleted. The only case where the tool will delete a tenant is when the tenant was created by our tool (identified by a unique tag), and has no other clusters provisioned under it.
+
+## 3. Mechanism
+
+acc-provision support 2 mode of handling tenant:
+1. It can create a dedicated tenant for the cluster
+2. A shared/pre-existing tenant when the tenant name is specified
+
+Depending on the mode the following behaviours are expected when the config is de-provisioned by using the `-d` option -
+
+1. The whole tenant is deleted
+2. Only the config added by acc-provision is deleted
+
+In case 1 scenario, initially if you run acc-provision with the -d option the tenant is not deleted if there is at least 1 application profile. However the tenant gets deleted in all other cases. e.g with pre-existing: VRFs, BDs, L3OUT and now if the "tenant name" is not specified in the acc provision input file the whole tenant should be deleted irrespective of number of application profile.
+
+In case 2 scenario, when it's pre-existing-tenant i.e when the tenant name is specified in the acc provision input file in aci_config section, the manually created tenant on the APIC before provisioning is not being deleted when unprovisioning i.e -d option but only the config added by provisioning i.e -a is deleted.
```

Comment:
Also any additional config that are not managed directly by acc-provision are not deleted